leshan-lwm2m-core/src/main/java/org/eclipse/leshan/senml/cbor/upokecenter/SenMLCborPackSerDes.java

```diff
@@ -68,98 +68,120 @@ public SenMLPack deserializeFromCbor(Collection<CBORObject> objects) throws SenM
     }
 
     protected SenMLRecord deserializeRecord(CBORObject o) throws SenMLException {
-        SenMLRecord record = new SenMLRecord();
-
-        deserializeBaseName(o, record);
-        deserializeBaseTime(o, record);
-        deserializeName(o, record);
-        deserializeTime(o, record);
-        boolean hasValue = deserializeValue(o, record);
-        hasValue |= deserializeBooleanValue(o, record);
-        hasValue |= deserializeStringValue(o, record);
-        hasValue |= deserializeObjectLinkValue(o, record);
-        hasValue |= deserializeOpaqueValue(o, record);
-
-        if (!allowNoValue && !hasValue) {
+
+        String recordBaseName = null;
+        BigDecimal recordBaseTime = null;
+        String recordName = null;
+        BigDecimal recordTime = null;
+        Number recordNumberValue = null;
+        Boolean recordBooleanValue = null;
+        String recordStringValue = null;
+        String recordObjectLinkValue = null;
+        byte[] recordOpaqueValue = null;
+
+        recordBaseName = deserializeBaseName(o);
+        recordBaseTime = deserializeBaseTime(o);
+        recordName = deserializeName(o);
+        recordTime = deserializeTime(o);
+        recordNumberValue = deserializeValue(o);
+        recordBooleanValue = deserializeBooleanValue(o);
+        recordStringValue = deserializeStringValue(o);
+        recordObjectLinkValue = deserializeObjectLinkValue(o);
+        recordOpaqueValue = deserializeOpaqueValue(o);
```

Comment:
here should merge both ? 

I mean instead of : 
```
String recordBaseName = null;
recordBaseName = deserializeBaseName(o);
```
we could write : 
```
String recordBaseName = deserializeBaseName(o);
```